New config options, Tab and Shift+Tab for moving caret

jtherrmann opened the pull request:
I've added two new config options that I find useful.

The first allows the user to prevent Instant Search from automatically opening a page when it is the only one found for the query. I find that with this behavior is enabled, I always end up accidentally editing the page because I'm still typing my query when it's opened (because I always leave the notebook in edit mode rather than read-only mode).

The second allows Instant Search to work with versions of Zim earlier than 0.66-. Debian Stretch uses Zim 0.65-4 and I'm still running Jessie anyway, so this seems like a useful option for saving Debian users from having to use an old Instant Search commit.

I've also added Tab for moving the caret down and Shift+Tab for moving the caret up. These keys are much more easily accessible for me than the up and down keys are.

Let me know if you have any concerns about these changes. I hope to be able to contribute to this awesome plugin. :)

jtherrmann committed:
Add config option for automatic page opening

Add config option to enable or disable automatically opening a page when
it is the only page found for the search query. Disabling this behavior
prevents accidentally modifying the page due to it opening automatically
while the user is still typing the query.

jtherrmann committed:
Add config option for older versions of Zim

Add config option to allow Instant Search to work with versions of Zim
earlier than 0.66-. See issues #24 and #25.

jtherrmann committed:
Update README

Update README to recommend using the config option for older versions of
Zim.

jtherrmann committed:
Add Tab and Shift+Tab for moving caret

Allow Tab to move caret down and Shift+Tab to move caret up

e3rd commented on instantsearch.py:
```diff
@@ -102,13 +104,32 @@ def instantsearch(self):
-                self.cached_titles.append((start2, start2.lower()))
-                build(start2)
+            if self.old_zim is False:
+                for s in self.window.ui.notebook.pages.list_pages(Path(start or ":")):
```
I think it would be better idea to determine the version change programatically. `old_zim` wouldn't be needed anymore.

```python
if pages in self.window.ui.notebook: # 
    o = self.window.ui.notebook.pages
else: # for zim0.66-
    o = self.window.ui.notebook.index
for s in o.list_pages(Path(start or ":")):
    ...
```

e3rd commented on instantsearch.py:
```diff
@@ -240,9 +261,10 @@ def startZimSearch(self):
-        if len(self.state.menu) == 1:
-            self._open_page(Path(self.state.menu.keys()[0]), excludeFromHistory=False)
-            self.close()
+        if self.open_when_unique is True:
```
let's put this on one line
```if self.open_when_unique and len(self.state.menu) == 1:```

e3rd reviewed (changes requested):
Thanks for the PR!
Can you please get rid of `old_zim` as requested in the comment and test?

jtherrmann committed:
Automatically work with Zim 0.66-

No longer require manual configuration depending on Zim version. Update
README to reflect my testing the plugin on Debian 8.9.

jtherrmann commented:
Good call. I've added the requested changes and tested successfully with Zim 0.62-3, 0.65-4, and 0.67-1.

e3rd reviewed (approved):
Fabulous!
